cmd/kube-proxy/app/server.go

```diff
@@ -252,7 +253,33 @@ func NewProxyServerDefault(config *ProxyServerConfig) (*ProxyServer, error) {
 		glog.V(2).Info("Using iptables Proxier.")
 		proxierIptables, err := iptables.NewProxier(iptInterface, execer, config.IptablesSyncPeriod, config.MasqueradeAll)
 		if err != nil {
-			glog.Fatalf("Unable to create proxier: %v", err)
+			iptables_err := err.Error()
+			iptables_err_str := ""
+			if strings.Contains(iptables_err, "iptables") {
+				//	Error is most probably caused by lack of iptables
+				// configuration
+				iptables_err_str = "Check if iptables has been enabled in the kernel. " +
```

Comment:
No the proxier will not be able to detect the kernel modules. As per the discussion on this issue I thought it was decided since there are too many variables to deal with (some of these modules might be compiled in kernel) it was better to throw a more elaborate error, as to the possible reason for failures of iptables, and leave it to the administrator to fix the issue. Hence the error string.